src/module_wrap.cc

```diff
@@ -987,17 +975,23 @@ Maybe<URL> ResolveExportsTarget(Environment* env,
         target_obj->GetOwnPropertyNames(context).ToLocalChecked();
     Local<Value> conditionalTarget;
     bool matched = false;
+    printf("CHECKING FOR ARRAY INDEX KEYS\n");
     for (uint32_t i = 0; i < target_obj_keys->Length(); ++i) {
+      printf("A\n");
       Local<Value> key =
           target_obj_keys->Get(context, i).ToLocalChecked();
+      printf("B\n");
       if (IsArrayIndex(env, key)) {
+        printf("CA\n");
         const std::string msg = "Invalid package config for " +
             pjson_url.ToFilePath() + ", \"exports\" cannot contain numeric " +
             "property keys.";
         node::THROW_ERR_INVALID_PACKAGE_CONFIG(env, msg.c_str());
         return Nothing<URL>();
       }
+      printf("CB\n");
     }
+    printf("DOING KEY RESOLUTION\n");
     for (uint32_t i = 0; i < target_obj_keys->Length(); ++i) {
       Local<String> key =
           target_obj_keys->Get(context, i).ToLocalChecked().As<String>();
```

Comment:
@guybedford Also dm’d you about this, but I’m pretty sure the `.As<String>()` call here isn’t valid because `GetOwnPropertyNames()` returns numeric indices as numbers

Reply:
It exactly was these, but there were a few of them to do. Turns out the test case exposed the bugs, not the code changes!